livepeer/studio
# api: Improve session lifecycle handling

**What does this pull request do? Explain your changes. (required)**

This is a bit massive change, but it completely rewrites the way that we handle livestream
sessions and their corresponding recordings on the API. 

The explanation here started to get too big so I wrote a doc instead: 
https://lvpr.link/live-rec-fact

**Specific updates (required)**
 - Made the "session merging" fields private and deprecated
 - Removed all "write" usage of them, keeping only the "read" usage for backward compatibility on existing streams
 - Added a logic to re-use previous transcoding sessions if they are recent and haven't been used (crash-loop protection)
 - Started writing the `isActive` field on child streams ("transcoding sessions") from `stream-info-service`
 - Used that `isActive` field to trigger recording webhooks and generation of VOD assets from them
 - Created new `none` state on the `recordingStatus` to differentiate in progress from failed recordings
 - Make sure to send `recording.ready` and create VOD assets when the `isActive` hadn't been updated by mapic

**How did you test each of these updates (required)**
 - Record a livestream
 - Make sure the session shows up on the stream page
 - Make sure it has the correct length
 - Make sure it has recording link and that it works
 - Make sure a VOD asset is generated from the recording
 - Kill `MistProcLivepeer` mid-stream
 - Make sure a new session shows up and all recordings are available in the end

**Does this pull request close any open issues?**
Implements API-7

**Screenshots (optional)**
![image](https://user-images.githubusercontent.com/1613383/223244629-f19311d2-aab3-4fe9-94e2-386ca91e8752.png)
This `n/a` is so much better than all of them being `In progress` as before :feelsgood:
We can also later change the frontend to only display the sessions that actually have something in them (but lets wait for catalyst recordings)

**Checklist**
